### ISSUE TITLE
regression (4.2.0+): type error when used in a typescript project

### ISSUE THREAD

**Issue by AviVahl** [rexxars/react-markdown#336]
When type-checking my (ts-based) project, I got:
```
node_modules/react-markdown/index.d.ts:11:34 - error TS7016: Could not find a declaration file for module 'remark-parse'. '/home/avi/projects/project/node_modules/remark-parse/index.js' implicitly has an 'any' type.
  Try `npm install @types/remark-parse` if it exists or add a new declaration (.d.ts) file containing `declare module 'remark-parse';`

11 import {RemarkParseOptions} from 'remark-parse'
                                    ~~~~~~~~~~~~~~


Found 1 error.
```
The definition file [references](https://github.com/rexxars/react-markdown/blob/v4.2.0/index.d.ts#L11) `remark-parse`, which doesn't have a `.d.ts` file (for the used v5).

Can be resolved by upgrading to `remark-parse@7`, as they added built-in typescript typings as well.

Thanks!

**Comment by rexxars** [comment]
Thanks for reporting. I can't update to remark-parse as it introduced breaking changes, but I've inlined the RemarkParseOptions for now to prevent this error, and published 4.2.2

**Comment by AviVahl** [comment]
Appreciate the quick release!
